src/Packagist/Api/Result/Package.php

```diff
@@ -147,9 +147,18 @@ public function getFavers()
     }
 
     /**
+     * @deprecated 1.6.0 Use getAbandoned() instead
      * @return bool
      */
     public function isAbandoned()
+    {
+        return (bool) $this->abandoned;
+    }
+
+    /**
+     * @return bool|string
```

Comment:
What about adding a link to the packagist doc stating the possible return type of this function ?

> The Packagist API can return a boolean or a string

Reply:
The API docs on their end are lean. I might reference the website’s source code instead.

Reply:
Okay, also, can you define an `UPGRADE.md` file to state all the deprecations ? It'll be easier to create releases then by looking at this file IMHO ;)

Reply:
I've had a look and I can't actually find anything obvious about this, but I created a test repo, submitted it to Packagist and marked it abandoned in favour of silverstripe/framework. You can check the API response from `https://repo.packagist.org/p/robbieaverill/packagist-test-repo.json:

```json
{
  "packages": {
    "robbieaverill/packagist-test-repo": {
      "dev-master": {
        "name": "robbieaverill/packagist-test-repo",
        "description": "",
        "keywords": [],
        "homepage": "",
        "version": "dev-master",
        "version_normalized": "9999999-dev",
        "license": [],
        "authors": [
          {
            "name": "Robbie Averill",
            "email": "robbie@averill.co.nz"
          }
        ],
        "source": {
          "type": "git",
          "url": "https://github.com/robbieaverill/packagist-test-repo.git",
          "reference": "a36cbacf393cdd49702d946c905bc913e16f495a"
        },
        "dist": {
          "type": "zip",
          "url": "https://api.github.com/repos/robbieaverill/packagist-test-repo/zipball/a36cbacf393cdd49702d946c905bc913e16f495a",
          "reference": "a36cbacf393cdd49702d946c905bc913e16f495a",
          "shasum": ""
        },
        "type": "library",
        "time": "2019-07-23T16:50:32+00:00",
        "abandoned": "silverstripe/framework",
        "uid": 3119283
      }
    }
  }
}
```

Reply:
Okay, do maybe add a description in the doc bloc of this fn which explains the behavior of this fn (ie why in some cases it returns a `bool` and why it can also return a string`). If `packagist` lacks of doc about it, I think you can mention your comment with the `json` response in the docblock (`@see` instruction) ;)

Reply:
Thanks, both suggestions implemented